test/smoke/src/helpers/setupEnvironmentHelper.ts

```diff
@@ -45,6 +45,49 @@ export class SetupEnvironmentHelper {
         SetupEnvironmentHelper.patchMetroConfig(workspacePath);
     }
 
+    public static  prepareHermesReactNativeApplication(workspaceFilePath: string, resourcesPath: string, workspacePath: string, appName: string, customEntryPointFolder: string, version?: string) {
+        let commandClean = path.join(workspacePath, "android", "gradlew") + " clean";
+        let commandBuild = path.join(workspacePath, "android", "gradlew") + " build";
+
+        console.log(`*** Executing  ${commandClean}...`);
+        cp.execSync(commandClean, { cwd: path.join(workspacePath, "android"), stdio: "inherit" });
+
+        let customEntryPointFile = path.join(resourcesPath, customEntryPointFolder, "App.js");
+        let launchConfigFile = path.join(resourcesPath, "launch.json");
+        let vsCodeConfigPath = path.join(workspacePath, ".vscode");
+        let appGradlBuildFilePath = path.join(workspacePath, "android", "app", "build.gradle");
+        let resGradlBuildFilePath = path.join(resourcesPath, customEntryPointFolder, "build.gradle");
+        let resReactGradlFilePath = path.join(resourcesPath, customEntryPointFolder, "react.gradle"); // must be removed after react-native package fix
+        let projReactGradlFilePath = path.join(workspacePath, "node_modules", "react-native", "react.gradle"); // must be removed after react-native package fix
+        let testButtonPath = path.join(resourcesPath, customEntryPointFolder, "TestButton.js");
+
+        console.log(`*** Copying  ${customEntryPointFile} into ${workspaceFilePath}...`);
+        fs.writeFileSync(workspaceFilePath, fs.readFileSync(customEntryPointFile));
+
+        console.log(`*** Copying  ${resGradlBuildFilePath} into ${appGradlBuildFilePath}...`);
+        fs.writeFileSync(appGradlBuildFilePath, fs.readFileSync(resGradlBuildFilePath));
+
+        console.log(`*** Copying  ${resReactGradlFilePath} into ${projReactGradlFilePath}...`); // must be removed after react-native package fix
+        fs.writeFileSync(projReactGradlFilePath, fs.readFileSync(resReactGradlFilePath));
+
+        console.log(`*** Copying ${testButtonPath} into ${workspacePath}`);
+        fs.copyFileSync(testButtonPath, path.join(workspacePath, "TestButton.js"));
+
+        if (!fs.existsSync(vsCodeConfigPath)) {
+            console.log(`*** Creating  ${vsCodeConfigPath}...`);
+            fs.mkdirSync(vsCodeConfigPath);
+        }
+
+        if (!fs.existsSync(path.join(vsCodeConfigPath, "launch.json"))) {
+            console.log(`*** Copying  ${launchConfigFile} into ${vsCodeConfigPath}...`);
+            fs.writeFileSync(path.join(vsCodeConfigPath, "launch.json"), fs.readFileSync(launchConfigFile));
+        }
+
+        console.log(`*** Executing  ${commandBuild}...`);
+        cp.execSync(commandBuild, { cwd: path.join(workspacePath, "android"), stdio: "inherit" });
```

Comment:
From my understanding we don't need to build an application before launch because the debug scenario will do this.